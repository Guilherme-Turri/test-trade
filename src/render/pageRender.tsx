import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../styles/theme';
import store from '../storage/store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

export const pageRender = (children: React.ReactNode) => {
  return render(
    <ThemeProvider theme={theme}>
      <Provider store={store}>
        <BrowserRouter>
          {children}
        </BrowserRouter>
      </Provider>
    </ThemeProvider>);
};