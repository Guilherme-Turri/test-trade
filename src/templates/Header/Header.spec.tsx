import { Header } from '.'
import { pageRender } from '../../render/pageRender'
import { fireEvent, screen } from '@testing-library/react'
import * as reactRedux from 'react-redux'


const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom') as any,
  useNavigate: () => mockedUsedNavigate,
}));

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
}));

describe('<Header />', () => {

  const useSelectorMock = reactRedux.useSelector as jest.Mock;
  const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch')

  it('should render on screen properly', () => {
    useSelectorMock.mockReturnValue({
      user: {
        name: 'testUser',
        apiKey: "123456",
      }
    });
    const { container } = pageRender(<Header />)
    const name = screen.getByText('OlÃ¡, testUser.')
    expect(name).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  })

  it('should call logout actions', () => {
    useSelectorMock.mockReturnValue({
      user: {
        name: 'testUser',
        apiKey: "123456",
      }
    });
    const dummyDispatch = jest.fn()
    useDispatchMock.mockReturnValue(dummyDispatch)

    pageRender(<Header />)
    const logoutBtn = screen.getByTestId('Logout')
    fireEvent.click(logoutBtn);
    expect(dummyDispatch).toHaveBeenCalled();
    expect(mockedUsedNavigate).toHaveBeenCalledWith('/');

  })
})