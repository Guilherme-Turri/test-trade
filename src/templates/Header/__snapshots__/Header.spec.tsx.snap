// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render on screen properly 1`] = `
.c1 {
  color: #2d5839;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0.938rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 3rem;
  background-color: white;
  border-bottom: 2px solid #00c555;
  font-weight: bold;
  -webkit-animation: iYrysg .3s;
  animation: iYrysg .3s;
}

.c2 {
  width: 2.188rem;
  height: 2.188rem;
  cursor: pointer;
}

.c4 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c3 {
  color: #00c555;
  width: 100%;
  height: 100%;
}

<div>
  <header
    class="c0"
    data-testid="HeaderContainer"
  >
    <p
      class="c1"
    >
      Ol√°, 
      testUser
      .
    </p>
    <header
      class="c2"
    >
      <div
        class="c3"
        data-testid="Logout"
      >
        <svg
          aria-hidden="true"
          class="c4"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M336 376V272H191a16 16 0 0 1 0-32h145V136a56.06 56.06 0 0 0-56-56H88a56.06 56.06 0 0 0-56 56v240a56.06 56.06 0 0 0 56 56h192a56.06 56.06 0 0 0 56-56zm89.37-104-52.68 52.69a16 16 0 0 0 22.62 22.62l80-80a16 16 0 0 0 0-22.62l-80-80a16 16 0 0 0-22.62 22.62L425.37 240H336v32z"
          />
        </svg>
      </div>
    </header>
  </header>
</div>
`;
